### ISSUE TITLE
Invalid Lua script evaluation

### ISSUE THREAD

**Issue by olizarevichroman** [StackExchange/StackExchange.Redis#2273]
Hi, i'm calling this simple Lua script passing Redis key to get string value, but key is passed to the `ARGV` collection.
Also i've noticed that my key passed twice, first as an argument and then as a key. Profiler log provided below

```
var script = "return redis.call('get', @settingIdKey)");
var preparedScript = LuaScript.Prepare(script);

var redisResult = await database.ScriptEvaluateAsync(script, new
{
    settingIdKey = (RedisKey)"my key",
});
```

Profiler log:

```
"SCRIPT" "LOAD" "return redis.call('get', ARGV[1])"
"EVAL" "return redis.call('get', ARGV[1])" "1" "my key" "my key"
[0 lua] "get" "my key"
```

Based on the [documentation](https://stackexchange.github.io/StackExchange.Redis/Scripting.html) it should treat it as a key

**Comment by mgravell** [comment]
Indeed, this is a side-effect of the current auto-parameterization implementation. If you want fine-grained control over the semantics, I suggest using the simple ScriptEvaluate API and passing the args/keys explicitly.

**Comment by olizarevichroman** [comment]
Is it supposed to be fixed? Or this "side-effect" is expected behavior?

**Comment by madelson** [comment]
Is this not just broken as-is?

**Comment by mgravell** [comment]
Suboptimal: yes. Broken: no. Unless I'm missing something, it is correctly routed and all values used are correct. Keys are *generally* short, so duplicating the keys isn't pathologically bad. But yes, it could be improved. Honestly, this whole API could be improved - I'll add to my backlog to review it.
